lamindb/models/_describe.py

```diff
@@ -127,81 +127,113 @@ def format_bytes(bytes_value):
         return f"{bytes_value / (1024**4):.1f} TB"
 
 
-def describe_artifact_general(
-    self: Artifact,
+def describe_dataset(
+    obj: Artifact | Collection,
     tree: Tree | None = None,
     foreign_key_data: dict[str, dict[str, int | str]] | None = None,
 ) -> Tree:
+    from lamindb.models import Artifact
+
     if tree is None:
-        tree = describe_header(self)
+        tree = describe_header(obj)
 
-    # add general information (order is the same as in API docs)
     general = tree.add(Text("General", style="bold bright_cyan"))
 
-    if self.key:
-        general.add(Text.assemble(("key: ", "dim"), (f"{self.key}", "cyan3")))
-    if self.description:
+    if obj.key:
+        general.add(Text.assemble(("key: ", "dim"), (f"{obj.key}", "cyan3")))
+    if obj.description:
         general.add(
             Text.assemble(
                 ("description: ", "dim"),
-                f"{self.description}",
+                f"{obj.description}",
             )
         )
 
-    # Two column items (short content)
     two_column_items = []
+    two_column_items.append(Text.assemble(("uid: ", "dim"), f"{obj.uid}"))
+
+    if isinstance(obj, Artifact):
+        two_column_items.append(Text.assemble(("hash: ", "dim"), f"{obj.hash}"))
+        two_column_items.append(
+            Text.assemble(("size: ", "dim"), f"{format_bytes(obj.size)}")
+        )
 
-    two_column_items.append(Text.assemble(("uid: ", "dim"), f"{self.uid}"))
-    two_column_items.append(Text.assemble(("hash: ", "dim"), f"{self.hash}"))
-    two_column_items.append(
-        Text.assemble(("size: ", "dim"), f"{format_bytes(self.size)}")
-    )
     transform_key = (
         foreign_key_data["run"]["transform_key"]
-        if foreign_key_data
-        else self.transform.key
-        if self.transform
+        if foreign_key_data and isinstance(obj, Artifact)
+        else foreign_key_data["transform"]["name"]
+        if foreign_key_data and "transform" in foreign_key_data
+        else obj.transform.key
+        if hasattr(obj, "transform") and obj.transform and isinstance(obj, Artifact)
```

Comment:
A main goal of the previous refactor was to get rid of the non-idiomatic `hasattr(obj, "transform")` calls. Imagine you'd have to do this for every type of sqlrecord.

And in addition to this being non-idiomatic it conceals performance considerations; Django makes database requests upon these checks. I think devs can learn this when they see `record.created_by` (because this is well known for Django), but for `hasattr(record, created_by)` it really surprised even me (shouldn't have) after multiple years of Django.